client/src/main/java/io/prometheus/client/metrics/builtin/JVMMetrics.java

```diff
@@ -0,0 +1,233 @@
+package io.prometheus.client.metrics.builtin;
+
+import com.google.common.collect.ImmutableMap;
+import io.prometheus.client.Register;
+import io.prometheus.client.Registry;
+import static io.prometheus.client.Registry.emptyLabels;
+import io.prometheus.client.metrics.Gauge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.JMException;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.*;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+/**
+ * <p>
+ *   Standard JVM metrics via MXBeans. These are registered automatically.
+ * </p>
+ *
+ * @author kim.altintop@gmail.com (Kim Altintop)
+ */
+public class JVMMetrics {
+  private static final Logger log = LoggerFactory.getLogger(JVMMetrics.class);
+  private static final Pattern WHITESPACE = Pattern.compile("[\\s]+");
+  private static final Pattern CAMEL_CASE = Pattern.compile("([a-z\\d])([A-Z])");
+
+  @Register(name = "memory_usage_heap_total", 
+      docstring = "Overall JVM heap memory usage", baseLabels = {})
+  private static final Gauge heapMemTotal = new Gauge();
+
+  @Register(name = "memory_usage_non_heap_total",
```

Comment:
Actually, can we have heap and non-heap in a single variable that has a label to differentiate the two? It then makes it easy to stack.